tests/exports-require-release-tags.js

```diff
@@ -0,0 +1,282 @@
+const path = require("path");
+const { RuleTester } = require("eslint");
+const BentleyESLintPlugin = require("../dist");
+const RequireReleaseTags =
+  BentleyESLintPlugin.rules["exports-require-release-tags"];
+
+const fixtureDir = path.join(
+  __dirname,
+  "fixtures",
+  "no-internal",
+  "workspace-pkg-1"
+);
+const tester = new RuleTester({
+  parser: require.resolve("@typescript-eslint/parser"),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: "module",
+    tsconfigRootDir: fixtureDir,
+    shouldCreateDefaultProgram: true,
+    project: path.join(fixtureDir, "tsconfig.test.json"),
+  },
+});
+
+tester.run("exports-require-release-tags", RequireReleaseTags, {
+  // extensions can only be public
+  valid: [
+    {
+      code: `
+      /**
+       * @extensions
+       * @public
+       */
+      export function destroyAllIModels(): void{
+      }
+      `,
+      options: [{ releaseTags: ["public", "beta"] }],
+    },
+    // test defaults
+    {
+      code: `
+        /**
+         * @extensions
+         * @public
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+    },
+    // non-extensions
+    {
+      code: `
+        /**
+         * @alpha
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+    },
+    {
+      code: `
+        /**
+         * @beta
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+    },
+    {
+      code: `
+        /**
+         * @public
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+    },
+    {
+      code: `
+        /**
+         * @internal
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+    },
+    // custom options for non-extensions
+    {
+      code: `
+        /**
+         * @alpha
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+      options: [{ releaseTags: ["public", "alpha"] }],
+    },
+    // empty options triggers default
+    {
+      code: `
+        /**
+         * @extensions
+         * @public
+         */
+        export function destroyAllIModels(): void{
+        }
+        `,
+      options: [{}],
+    },
+  ],
+  invalid: [
+    // extensions require public tag.
+    {
+      code: `
+        /**
+         * @extensions
+         */
+        export function destroyAllIModels(): void {
+
+        }
+        `,
+      errors: [
+        {
+          message: `Extension exports must be annotated with both an @extensions and @public tag. Check the "destroyAllIModels" type.`,
```

Comment:
not sure I understand what this means. isn't `"destroyAllIModels"` a function and not a type?

Reply:
Everything in TS has a type, so I went with that, but maybe it's enough to just say `destroyAllIModels`

Reply:
Did you mean `Check type of "destroyAllIModels"`? because `destroyAllIModels` isn't name of a type in core repo or anything like that right? correct me if I am wrong

Reply:
Right no, it's just a made up function for the test case. 
I did mean `check type of {{}}` I think that makes more sense.

Reply:
Ah, I had hard coded `destroyAllIModels`... anyway, reworded

Reply:
I think I would prefer something like `Please review the tags for "destroyAllIModels" for reference.` or `Please review the tags for "destroyAllIModels" for example.` but the way it is I am not opposed to it either. Thanks for changing